open-craft/learner-profile-dashboard
# Upgrade to Django 1.11

Cf. [SE-885](https://tasks.opencraft.com/browse/SE-885)

**Dependencies**

https://github.com/open-craft/django-lti-tool-provider/pull/9

**Test instructions**

1. Create venv:
    ```sh
    virtualenv test-env
    source test-env/bin/activate
    ```

2. Check out learner-profile-dashboard repo and switch to this PR's feature branch:
    ```sh
    git clone git@github.com:open-craft/learner-profile-dashboard.git
    cd learner-profile-dashboard
    git checkout origin/deps-update
    ```

3. Install dependencies, making sure that `pip` doesn't complain about any version conflicts:
    ```sh
    pip install -r requirements.txt
    ```

4. Create `app/local_settings.py`:
    ```sh
    touch app/local_settings.py
    ```
    For the purpose of testing this PR it's not necessary to override default settings in `app/local_settings.py`. We're just creating this file to silence [this warning](https://github.com/open-craft/learner-profile-dashboard/blob/845f33ed3d5913c0e78f3b8d4dc48d5b38f6be11/app/settings.py#L228-L235).

4. Run migrations:
    ```sh
    ./manage.py migrate
    ```

5. Run tests, making sure that the output contains no deprecation warnings:
    ```sh
    python -Wa manage.py test
    ```

**Author notes**

- [x] Will need to update django-lti-tool-provider reference in `requirements/_base.txt` after merging https://github.com/open-craft/django-lti-tool-provider/pull/9 and tagging the latest version.

**Reviewers**
